src/modules/api/user/user.service.ts

```diff
@@ -2,8 +2,13 @@ import { Injectable } from '@nestjs/common';
 import { InjectModel } from '@nestjs/mongoose';
 import { User } from 'src/schemas/user.schema';
 import { Model } from 'mongoose';
+import { async } from 'rxjs';
 
 @Injectable()
 export class UserService {
   constructor(@InjectModel(User.name) private userModel: Model<User>) {}
+
+  async getUserById(id) {
+    return await this.userModel.findById(id);
```

Comment:
Mention type of ID.

Reply:
Changed to Model.ObjectId

Reply:
It should be string

Reply:
i dont think we can use findById on a string ?

Reply:
I think we can. And either way, it would make sense to construct ObjectId thru a string here, rather than controller.